### DISCUSSION TITLE
Why installing on Windows is so complicated?

### DISCUSSION THREAD

**Discussion by Edremelech** [facebookresearch/pytorch3d#1007] [General]
Hello guys,
I've just chosen the project for my diploma, and it requires PyTorch3d. Guess how I was disappointed when I was **unable** to install it? This is the first time I can't install a Python library (OK, not the first, but the first library wasn't so popular and significant).
Why the Linux users can just install the prebuilt package, but the hapless Windows users have to "go to the "x64 Native Tools Command Prompt for VS 2019" and build manually? And "make some changes in the headers" before compiling that I should find **somewhere** (they aren't even listed on the installation page)? What if I don't have VS2019? It isn't free BTW.

Do you really hate Bill Gates that much? I don't like him much myself, but I still use Windows. What to do now?

I have PyTorch 1.10 and CUDA 10.2 (at least torch.version yields '1.10.0+cu102'). Please spare me from the suggestions like "use Colab". I've bought a video card specifically to **avoid** using Colab.

**Comment by 3DAlgoLab** [comment]
In WSL2, it works with no problem. But I also hope pytorch3d support windows natively(?). I tried several ways to install pytorch3d in windows through building source. But most trials were not sucessful. Wihtout cuda supports, it can be installed easily. But I think it's not meaningful in that way. If anyone here succeeded in installing in windows with CUDA supporting, plz share some recipes. Thanks in advance.
